# Random Forest Classifier carbon, silicon, nitrogen, and aluminum isotopes

In [9]:
#Imports
import joblib
import pandas as pd

from sklearn import tree
from sklearn.ensemble import RandomForestClassifier
from sklearn.model_selection import GridSearchCV
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import LabelEncoder, MinMaxScaler


In [10]:
# Read csv file
C_Si_N_Al = pd.read_csv('presolargrains_C_Si_N_Al.csv')
C_Si_N_Al.head()

Type  carbon_12_13  nitrogen_14_15  aluminum_26_27  silicon_29_28  \
0    X       1581.00           116.0          0.0095         -684.0   
1    X        234.75           187.2          0.3327         -683.0   
2    X        140.00            97.0          0.0170         -653.0   
3    X        223.00           102.0          0.1140         -600.0   
4    X       1693.89            63.2          0.2364         -588.0   

   silicon_30_28  
0         -490.0  
1         -501.0  
2         -446.0  
3         -459.0  
4         -605.0

In [12]:
# Set up data and feature names; display dataframe dropping columns
data = C_Si_N_Al.drop("Type", axis=1)
feature_names = data.columns
data.head()

carbon_12_13  nitrogen_14_15  aluminum_26_27  silicon_29_28  silicon_30_28
0       1581.00           116.0          0.0095         -684.0         -490.0
1        234.75           187.2          0.3327         -683.0         -501.0
2        140.00            97.0          0.0170         -653.0         -446.0
3        223.00           102.0          0.1140         -600.0         -459.0
4       1693.89            63.2          0.2364         -588.0         -605.0

In [13]:
target = C_Si_N_Al['Type']

In [14]:
# Separate data into train and test buckets
X_train, X_test, y_train, y_test = train_test_split(data, target, random_state=1)

## Scale the data using MinMaxScalar

In [15]:
X_scaler = MinMaxScaler().fit(X_train)
X_train_scaled = X_scaler.transform(X_train)
X_test_scaled = X_scaler.transform(X_test)

## Train the model; run the test score

In [16]:
# Random Forest Classifier
rf = RandomForestClassifier(n_estimators=200)
rf = rf.fit(X_train_scaled, y_train)
rf.score(X_test_scaled, y_test)

0.9512195121951219

In [17]:
# Save fitted model to file
filename = 'RandomForestCls_C_Si_N_Al.sav'
joblib.dump(rf, filename)

['RandomForestCls_C_Si_N_Al.sav']

# Summary:

## Random Forest Classifier carbon, silicon, nitrogen, and aluminum :  95.1%